### PR TITLE
Feature 99: Redux for Boilers

### PR DESCRIPTION
**Pull request by ValentinSilvestri**
- [x] Get all boilers works.
- [x] Delete a boiler works.
- [x] Update a boiler works. 
- [x] Add new boiler.

_Features to develop:_
- Add validations of the inputs on the FrontEnd. 
- Add alert messages if a boiler it's in use or a boiler type doesn't exist.
- Select button on boiler types with the IDs in the DB.

### CODE REVIEW

File: src/components/Boilers/BoilersForm/index.js [hunk 1]
```diff
@@ -10,51 +9,33 @@ const BoilersForm = ({
   boiler,
 }) => {
   const [state, setState] = useState({
-    description: boiler.description || 'Boiler ...',
-    boilerType: boiler.boilerType || 'boilerTypeId1',
+    description: boiler.description || 'Boiler Nº ....',
```
Comment (lmalvasia) [inline src/components/Boilers/BoilersForm/index.js:12]:
```suggestion
    description: boiler.description || '',
```

File: src/components/Boilers/BoilersForm/index.js [hunk 2]
```diff
@@ -63,22 +44,24 @@ const BoilersForm = ({
           <input type="text" id="description" name="description" value={state.description} onChange={onChangeInput} />
         </div>
         <div className={styles.inputContainer}>
-          <Select label="Boilers types:" name="boilerType" value={state.boilerType} onChange={onChangeInput} options={boilerTypes} />
+          <label htmlFor="boilerType">Boiler type</label>
+          <input type="text" id="boilerType" name="boilerType" value={state.boilerType} onChange={onChangeInput} />
```
Comment (lmalvasia) [inline src/components/Boilers/BoilersForm/index.js:48]: This should be a select field. We can continue using text input but we should change it for the next week.

File: src/components/Boilers/index.js [hunk 3]
```diff
@@ -59,8 +62,9 @@ const Boilers = () => {
             </tr>
           </thead>
           <tbody>
-            {boilers.map((boiler) => (
-              <tr key={boiler.id}>
+            {boilers.list.map((boiler) => (
+              // eslint-disable-next-line
```
Comment (lmalvasia) [inline src/components/Boilers/index.js:66]: Why you are adding this comment? We should remove the eslint rule if it's not necessary.

File: src/components/Boilers/index.js [hunk 4]
```diff
@@ -87,7 +91,8 @@ const Boilers = () => {
                       show: true,
                       type: 'DELETE',
                       meta: {
-                        id: boiler.id,
+                        // eslint-disable-next-line
```
Comment (lmalvasia) [inline src/components/Boilers/index.js:94]: Same as above.

File: src/components/Boilers/index.js [hunk 5]
```diff
@@ -97,6 +102,7 @@ const Boilers = () => {
             ))}
           </tbody>
         </table>
+        {boilers.loading ? <span>LOADING...</span> : null}
```
Comment (lmalvasia) [inline src/components/Boilers/index.js:105]: We should render this message or the table, not both.

File: src/redux/actions/boilersAction.js [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+export const deleteBoiler = (id) => async (dispatch) => {
+  dispatch(deleteBoilerFetching());
+  try {
+    const data = await fetch(`${URL}/${id}`, {
+      method: 'DELETE',
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
+    if (data.status === 400) {
```
Comment (lmalvasia) [inline src/redux/actions/boilersAction.js:72]: These validations are not fine.
You can do something like:
```
if (data.ok) {
  dispatch(deleteBoilerFulfilled(id));
} else {
  throw new Error('Error');
}
```

File: src/redux/actions/boilersAction.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+    const data = await fetch(URL, {
+      method: 'POST',
+      body: JSON.stringify(boiler),
+      headers: {
+        'Content-type': 'application/json',
+      },
+    });
+    //const res = await data.json();
+    //return dispatch(addBoilerFulfilled());
+    return dispatch(getBoilers());
```
Comment (lmalvasia) [inline src/redux/actions/boilersAction.js:113]: This is not ok. You have to get the data from the response and dispatch addBoilerFulfilled.

File: src/redux/reducers/boilersReducer.js [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+      };
+    case ADD_BOILER_FETCHING:
+      return {
+        ...state,
+        loading: true,
+      };
+    case ADD_BOILER_FULFILLED:
+      return {
+        ...state,
+        list: [...state.list, action.payload.building],
```
Comment (lmalvasia) [inline src/redux/reducers/boilersReducer.js:50]:
```suggestion
        list: [...state.list, action.payload],
```

File: src/redux/actions/boilersAction.js [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+  }
+};
+
+const addBoilerFetching = () => ({
+  type: ADD_BOILER_FETCHING,
+});
+
+const addBoilerFulfilled = (boiler) => ({
+  type: ADD_BOILER_FULFILLED,
+  payload: {
```
Comment (lmalvasia) [inline src/redux/actions/boilersAction.js:92]:
```suggestion
  payload: boiler
```

File: src/redux/reducers/boilersReducer.js [hunk 10]
```diff
@@ -0,0 +1,110 @@
...
+      return {
+        ...state,
+        loading: true,
+      };
+    case UPDATE_BOILER_FULFILLED:
+      return {
+        ...state,
+        loading: false,
+        list: state.list.map((boiler) => {
+          // eslint-disable-next-line
```
Comment (lmalvasia) [inline src/redux/reducers/boilersReducer.js:68]: Remove this comment and remove the eslint rule.

**Review by lmalvasia — CHANGES_REQUESTED** [review]

File: src/components/Boilers/BoilersForm/index.js (hunk 2 above)
Comment (ValentinSilvestri) [reply src/components/Boilers/BoilersForm/index.js:48]: Ok, it will be a select field for the next week!

**Review by lmalvasia — APPROVED** [review]

**Pull request merged**